WTG.Analyzers/Analyzers/BooleanLiteral/ArgumentListExtensions.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace WTG.Analyzers
+{
+	static class ArgumentListExtensions
+	{
+		public static IParameterSymbol? TryFindCorrespondingParameterSymbol(this ArgumentListSyntax argumentList, int index, SemanticModel semanticModel, CancellationToken cancellationToken)
+		{
+			var methodInvocation = argumentList.FirstAncestorOrSelf<InvocationExpressionSyntax>();
+			if (methodInvocation is null)
+			{
+				return null;
+			}
+
+			var methodSymbol = (IMethodSymbol)semanticModel.GetSymbolInfo(methodInvocation, cancellationToken).Symbol;
+			if (methodSymbol is null)
+			{
+				return null;
+			}
+
+			var argumentSymbol = methodSymbol.Parameters[index];
+			return argumentSymbol;
+		}
+
+		public static int FindIndexOfArgument(this ArgumentListSyntax haystack, ArgumentSyntax needle)
```

Comment:
haystack? needle? feels like the names from a unit test :)

Reply:
I stole them from PHP actually... not usually the best idea when it comes to names, but I feel it makes it clear which object you're looking _in_ and which object you're looking _for_.